baseband/gsb/base.py

```diff
@@ -522,49 +526,55 @@ def open(name, mode='rs', **kwargs):
-    nchan : int, optional
-        Number of channels. Default: 1 for rawdump, 512 for phased.
-    bps : int, optional
-        Bits per elementary sample (e.g., the real or imaginary part of each
-        complex data sample).  Default: 4 for rawdump, 8 for phased.
-    complex_data : bool, optional
-        Default: `False` for rawdump, `True` for phased.
     samples_per_frame : int
         Number of complete samples per frame.  Can give ``payloadsize``
         instead.
     payloadsize : int, optional
         Number of bytes per payload, divided by the number of raw files.
-        If both ``samples_per_frame`` and ``payloadsize`` are ``None``,
-        ``payloadsize`` is set to 2**22 (4 MB) for rawdump and 2**23 (8 MB)
+        If both ``samples_per_frame`` and ``payloadsize`` are `None`,
+        ``payloadsize`` is set to 2**22 (4 MB) for rawdump, and 2**23 (8 MB)
         divided by the number of streams per polarization for phased.
+    nchan : int, optional
+        Number of channels. Default: `None`, which sets it to 1 for rawdump,
+        512 for phased.
+    bps : int, optional
+        Bits per elementary sample (i.e. real or imaginary parts of a complex
+        sample).  Default is `None`, which sets it to 4 for rawdump, 8 for
+        phased.
+    complex_data : bool, optional
+        Whether data is complex.  Default: `False` for rawdump, `True` for
+        phased.
+    squeeze : bool, optional
+        If `True` (default) and reading, remove any dimensions of length unity
+        from decoded data.  If `True` and writing, accept squeezed arrays as
+        input, and adds any dimensions of length unity.
+
+    --- For reading only:  (see `~baseband.gsb.base.GSBStreamReader`)
+
     subset : indexing object or tuple of objects, optional
         Specific components of the complete sample to decode.  If a single
         indexing object is passed, it selects (available) polarizations.  If a
         tuple of objects is passed, the first selects (available) polarizations
         and the second selects channels.  By default, all components are read.
-    squeeze : bool, optional
-        If `True` (default) and reading, remove any dimensions of length unity
-        from decoded data.  If `True` and writing, accept squeezed
-        arrays as input, and add channel and polarization dimensions if they
-        have length unity.
 
-    --- For writing a stream : (see `~baseband.gsb.base.GSBStreamWriter`)
+    --- For writing only : (see `~baseband.gsb.base.GSBStreamWriter`)
 
     header : `~baseband.gsb.GSBHeader`
-        Header for the first frame, holding time information, etc.
+        Header for the first frame, holding time information, etc.  Can instead
+        give keyword arguments to construct a header.
     **kwargs
         If the header is not given, an attempt will be made to construct one
         with any further keyword arguments.  If one requires to explicitly set
```

Comment:
Maybe follow other formats and have a line with
```
--- Header Keywords ...
```
I would only put `header_mode` and `time` there...

Reply:
The only format that has `Header Keyworlds` in `open` is DADA, because the header is built in the opener rather than in `DADAStreamWriter`.